internal/errors/router.go

```diff
@@ -56,3 +60,37 @@ func (e unrecognizedProcedureError) Error() string {
 func (e unrecognizedProcedureError) AsHandlerError() HandlerError {
 	return HandlerBadRequestError(e)
 }
+
+// UnrecognizedEncodingError indicates that a request could not be handled locally because
+// the router contained no handler for the request's specific encoding.
+type UnrecognizedEncodingError interface {
+	error
+
+	unrecognizedEncodingError()
+}
+
+// RouterUnrecognizedEncodingError returns an error indicating that the router
+// could find no corresponding handler for the request's specific encoding.
+func RouterUnrecognizedEncodingError(want []string, got string) error {
+	return unrecognizedEncodingError{
+		Want: want,
+		Got:  got,
+	}
+}
+
+type unrecognizedEncodingError struct {
+	Want []string
+	Got  string
+}
+
+// brands this uniquely as an unrecognized encoding error.
+func (unrecognizedEncodingError) unrecognizedEncodingError() {}
```

Comment:
should an unrecognizedEncoding error be a subset of the UnrecognizedProcedureError?